### ISSUE TITLE
`is_serializable` returns False for "hello world" flow

### ISSUE THREAD

**Issue by orcaman** [PrefectHQ/prefect#1262]
Hi guys,

I have an issue on my flow that seems like it may be related to serialization. 

So I've tried using the is_serializable function, which returned `False` for my flow. I have made my flow simpler to test it and kept getting `False`, until I tried the simple "Hello World" example from the homepage, and was surprised to see that that one too returns `False` for the flow:

```python
from prefect import task, Flow
from prefect.utilities.debug import is_serializable

@task
def say_hello():
    print("Hello, world!")

with Flow("My First Flow") as flow:
    say_hello()

if is_serializable(flow) is False:
    raise ValueError('flow is not serializable at checkpoint 1')
    
flow.run() # "Hello, world!"
```

The output of the above is:
```
Traceback (most recent call last):
  File "hello.py", line 12, in <module>
    raise ValueError('flow is not serializable at checkpoint 1')
ValueError: flow is not serializable at checkpoint 1
```

Is this expected? 

Thanks

**Comment by joshmeek** [comment]
Hey @orcaman! So I ran your snippet and it worked properly for me:
```
[2019-07-24 13:04:16,880] INFO - prefect.FlowRunner | Beginning Flow run for 'My First Flow'
[2019-07-24 13:04:16,881] INFO - prefect.FlowRunner | Starting flow run.
[2019-07-24 13:04:16,885] INFO - prefect.TaskRunner | Task 'say_hello': Starting task run...
[2019-07-24 13:04:16,885] WARNING - prefect.TaskRunner | Task 'say_hello': can't use cache because it is now invalid
Hello, world!
[2019-07-24 13:04:16,886] INFO - prefect.TaskRunner | Task 'say_hello': finished task run for task with final state: 'Success'
[2019-07-24 13:04:16,886] INFO - prefect.FlowRunner | Flow run SUCCESS: all reference tasks succeeded
```

Not sure if it would have something to do with it but what version of prefect are you using?

**Comment by cicdw** [comment]
Hi @orcaman - this is not expected; could you share the version of Prefect you're on and some details about your Operating System (Windows vs. Mac vs. Linux)?

**Comment by orcaman** [comment]
Thanks. I'm on MacOS Mojave 10.14.5 (18F132) with prefect-0.6.0 and Python 3.7.3

**Comment by cicdw** [comment]
Very very interesting - so far I can't recreate.  Here's what I've tried:
- running your code interactively in an IPython session
- running your code interactively in a python session
- running your code as a script

Any chance you're running this code in some other way that I'm not capturing?

Ultimately the `is_serializable` utility is _intended_ to be helpful - it isn't actually called by Prefect at any time.  If you are running into serialization issues when running a "real" workflow, your problems could be related to the issues addressed in this new tutorial: https://docs.prefect.io/guide/tutorials/task-guide.html#task-inputs-and-outputs

**Comment by orcaman** [comment]
Thanks! 

I can't think of anything special that I'm doing. Anyways this isn't really critical for me, I found the problem since :-) 

You can feel free to close this if you like (I'm not going to need this functionality any time soon, hopefully).

**Comment by cicdw** [comment]
Ok glad to hear you solved it!  I'll leave this open for a while in case others come across a similar situation.

**Comment by cbowdon** [comment]
I came across this issue, it was really confusing. It turned out to be because the subprocess that is launched to very that `cloudpickle` can deserialize the flow in a separate process was not being launched with the same Python version and virtual environment as my Prefect project. I think this line is the problem:

https://github.com/PrefectHQ/prefect/blob/master/src/prefect/utilities/debug.py#L53

I worked around it by copying the source of `is_serializable` and editing the subprocess command, but a smarter solution would be look up the path of the current executable and use that.

**Comment by cicdw** [comment]
Awesome, great catch @cbowdon - let me know if you're interested in submitting a PR for the fix!  Reopening to track...

**Comment by cbowdon** [comment]
So this only occurs if you're calling a `python` that's not at the top of your `$PATH`, e.g. if for some reason you cannot activate your virtual environment and are running Prefect by calling the environment's `python` directly. Probably not a common use case. Using `sys.executable` rather than hard-coding `python` would most likely fix this, if I have time I will submit a PR to that effect but no promises I'm afraid @cicdw .

**Comment by jcrist** [comment]
Fixed by #2931. Closing.
